### PR TITLE
Distance curve, use weighting for straighten command, fix formatting

### PR DESCRIPTION
**Pull request by ryanlarkin**

### CODE REVIEW

**Comment by ounsworth** [comment]
I'm curious what you're using for distance curve. Could you link me to the
file and line? What's the reasoning behind it?

**Comment by ryanlarkin** [comment]
@ounsworth, I made a graph that should replicate what we do in this command [https://www.desmos.com/calculator/mqp23vqpjh](https://www.desmos.com/calculator/mqp23vqpjh).
`a` is calculated in [DriveWithDecelerationCurve @ 53](https://github.com/eandr127/2017-2706-Robot-Code/blob/distance-curve/src/org/usfirst/frc/team2706/robot/commands/autonomous/movements/DriveWithDecelerationCurve.java#L53)
`driveSpeed` (or y for graph) is calculated in [DriveWithDecelerationCurve @ 62](https://github.com/eandr127/2017-2706-Robot-Code/blob/distance-curve/src/org/usfirst/frc/team2706/robot/commands/autonomous/movements/DriveWithDecelerationCurve.java#L62)
`distanceFromObstacle` (or x in graph) is calculated in [DriveWithDecelerationCurve @ 62](https://github.com/eandr127/2017-2706-Robot-Code/blob/distance-curve/src/org/usfirst/frc/team2706/robot/commands/autonomous/movements/DriveWithDecelerationCurve.java#L62)

For the graph, c is the max speed, f is the min speed, d is the distance, s is the distance to start driving at base speed.

The original issue is #27. I think the reason we're doing this is so that the robot will stop by the time we reach our setpoint, and won't be going back and forth until it get's close enough.

File: src/org/usfirst/frc/team2706/robot/commands/autonomous/movements/DriveWithDecelerationCurve.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+     * @param maximumDistanceFromObstacle The range where the robot stops when the distance to
+     *        target is within
+     * @param minimumSpeed The minimum speed that the robot will drive at
+     * @param maximumSpeed The fastest the robot will drive at
+     * @param alpha The weight given to the current sensor data, verses the previous weighted
+     *        average
+     */
+    public DriveWithDecelerationCurve(double desiredDistanceToObstacle,
+                    double maximumDistanceFromObstacle, double minimumSpeed, double maximumSpeed,
+                    double alpha) {
```
Comment (cnnradams) [inline src/org/usfirst/frc/team2706/robot/commands/autonomous/movements/DriveWithDecelerationCurve.java]: Invert value

**Review by cnnradams — COMMENTED** [review]
I think we should wait on this pull request until the slow down thing has a useful purpose, maybe with the new braking robot.

**Comment by ryanlarkin** [comment]
I'll just close this for now because we have other priorities right now. We might come back to this later and reopen it.

**Pull request closed without merging**